### DISCUSSION TITLE
Clarification of job dependencies.  i.e. test -> needs: build

### DISCUSSION THREAD

**Discussion by ghstahl** [community/community#26144] [Actions]
<p><a href="https://github.com/fluffy-bunny/typescript-action-needs/actions" rel="nofollow noopener">https://github.com/fluffy-bunny/typescript-action-needs/actions</a></p>
<p>I started with the typescript-action template and made 2 modificaitons.</p>
<ol>
<li>Introduce needs: build to the test job</li>
</ol>
<p>I wanted to test the dist/index.js that the build job produces.</p>
<pre><code>ncc: Version 0.20.5
ncc: Compiling file index.js
12kB dist/index.js
12kB [654ms] - ncc 0.20.5
</code></pre>
<ol start="2">
<li>deleted ./dist/index.js</li>
</ol>
<p>result: </p>
<pre><code>Run ./
##[error]File not found: '/home/runner/work/typescript-action-needs/typescript-action-needs/./dist/index.js'
</code></pre>
<p>The build job produced the dist/index.js, but  the test job is not seing it.</p>
<p>Why?</p>

**Comment by airtower-luna** [comment]
<p>If those are separate jobs they will run on separate runners, so the “test” job does not (automatically) get the data from the “build” job.</p>
<p>Depending on your workflow it may make sense to use one job that does both, or you could use cache or artifact actions to pass data between jobs.</p>

**Comment by ghstahl** [comment, marked as answer]
<p><a href="https://github.com/fluffy-bunny/typescript-action-needs/blob/master/.github/workflows/test.yml" rel="nofollow noopener">https://github.com/fluffy-bunny/typescript-action-needs/blob/master/.github/workflows/test.yml</a></p>
<pre><code>name: "build-test"
on: # rebuild any PRs and main branch changes
  pull_request:
  push:
    branches:
      - master
      - 'releases/*'

jobs:
  build: # make sure build/ci work properly
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v1
    - run: |
        npm install
        npm run all

    - name: upload-artifact
      uses: actions/upload-artifact@v1
      with:
        name: my-artifact
        path: dist

  test: # make sure the action works on a clean machine without building
    needs: build
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v1

    - name: download-artifact from build
      uses: actions/download-artifact@v1
      with:
        name: my-artifact
        path: dist

    - uses: ./
      with:
        milliseconds: 1000
</code></pre>
<p>upload-&gt;download artifact works for my use case.</p>
<p>Thank you </p>
